nilearn/decoding/tests/test_space_net.py

```diff
@@ -0,0 +1,324 @@
+    rng = check_random_state(42)
+    X_test = rng.randn(n_samples, n_features)
+    y_test = np.dot(X_test, np.ones(n_features))
+    w = np.zeros(n_features)
+    escb = _EarlyStoppingCallback(X_test, y_test, False)
+    for counter in range(50):
+        k = min(counter, n_features - 1)
+        w[k] = 1
+
+        # jitter
+        if k > 0 and rng.rand() > .9:
+            w[k - 1] = 1 - w[k - 1]
+
+        escb(dict(w=w, counter=counter))
+        assert_equal(len(escb.test_scores), counter + 1)
+
+        # restart
+        if counter > 20:
+            w *= 0.
+
+
+def test_params_correctly_propagated_in_constructors():
+    for (penalty, is_classif, n_alphas, l1_ratio, n_jobs,
+         cv, perc) in itertools.product(
+        ["graph-net", "tv-l1"], [True, False],
+        [.1, .01], [.5, 1.], [1, -1], [2, 3], [5, 10]):
+        cvobj = BaseSpaceNet(
+            mask="dummy", n_alphas=n_alphas, n_jobs=n_jobs, l1_ratios=l1_ratio,
+            cv=cv, screening_percentile=perc, penalty=penalty,
+            is_classif=is_classif)
+        assert_equal(cvobj.n_alphas, n_alphas)
+        assert_equal(cvobj.l1_ratios, l1_ratio)
+        assert_equal(cvobj.n_jobs, n_jobs)
+        assert_equal(cvobj.cv, cv)
+        assert_equal(cvobj.screening_percentile, perc)
+
+
+def testlogistic_path_scores():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X_, mask = to_niimgs(X, [2, 2, 2])
+    mask = mask.get_data().astype(np.bool)
+    alphas = [1., .1, .01]
+    test_scores, best_w = logistic_path_scores(
+        _graph_net_logistic, X, y, mask, alphas, .5,
+        range(len(X)), range(len(X)), {})[:2]
+    test_scores = test_scores[0]
+    assert_equal(len(test_scores), len(alphas))
+    assert_equal(X.shape[1] + 1, len(best_w))
+
+
+def test_squared_loss_path_scores():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X_, mask = to_niimgs(X, [2, 2, 2])
+    mask = mask.get_data().astype(np.bool)
+    alphas = [1., .1, .01]
+    test_scores, best_w = squared_loss_path_scores(
+        _graph_net_squared_loss, X, y, mask, alphas, .5,
+        range(len(X)), range(len(X)), {})[:2]
+    test_scores = test_scores[0]
+    assert_equal(len(test_scores), len(alphas))
+    assert_equal(X.shape[1] + 1, len(best_w))
+
+
+# def test_alpha_attrs():
+#     iris = load_iris()
+#     X, y = iris.data, iris.target
+#     alpha = 1.
+#     X, mask = to_niimgs(X, (2, 2, 2))
+#     for penalty, is_classif, verbose in itertools.product(
+#             ['graph-net', 'tv-l1'], [True, False], [True, False]):
+#         cv_class = eval('SpaceNet%s' % (
+#             ['Regressor', 'Classifier'][is_classif]))
+#         cv = cv_class(
+#             mask=mask, penalty=penalty, alphas=alpha, verbose=verbose,
+#         ).fit(X, y)
+#         if is_classif:
+#             np.testing.assert_array_equal([alpha] * 3, cv.alphas_)
+#         else:
+#             np.testing.assert_array_equal([alpha], cv.alphas_)
```

Comment:
Rmed the commented code. Rmed mean_img stuff too.

On Sat, Jul 18, 2015 at 6:11 PM, Gael Varoquaux <notifications@github.com>
wrote:

> In nilearn/decoding/tests/test_space_net.py
> <https://github.com/nilearn/nilearn/pull/657#discussion_r34949090>:
>
> > +# def test_alpha_attrs():
> > +#     iris = load_iris()
> > +#     X, y = iris.data, iris.target
> > +#     alpha = 1.
> > +#     X, mask = to_niimgs(X, (2, 2, 2))
> > +#     for penalty, is_classif, verbose in itertools.product(
> > +#             ['graph-net', 'tv-l1'], [True, False], [True, False]):
> > +#         cv_class = eval('SpaceNet%s' % (
> > +#             ['Regressor', 'Classifier'][is_classif]))
> > +#         cv = cv_class(
> > +#             mask=mask, penalty=penalty, alphas=alpha, verbose=verbose,
> > +#         ).fit(X, y)
> > +#         if is_classif:
> > +#             np.testing.assert_array_equal([alpha] * 3, cv.alphas_)
> > +#         else:
> > +#             np.testing.assert_array_equal([alpha], cv.alphas_)
>
> Should this commented code be deleted?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/657/files#r34949090>.
>



-- 
DED